lib/webdriveragent.js

```diff
@@ -286,6 +300,24 @@ class WebDriverAgent {
       return await this.getStatus();
     }
 
+    if (this.isRealDevice && this.usePreinstalledWDA) {
+      const xctestEnv = {
+        USE_PORT: this.wdaLocalPort || WDA_AGENT_PORT,
+        WDA_PRODUCT_BUNDLE_IDENTIFIER: this.bundleIdForXctest
+      };
+      if (this.mjpegServerPort) {
+        xctestEnv.MJPEG_SERVER_PORT = this.mjpegServerPort;
+      }
+      this.log.info('Launching WebDriverAgent on the device without xcodebuild');
+      this.xctestSessionForWDA = new Xctest(this.device.udid, this.bundleIdForXctest, null, {env: xctestEnv});
+
+      await this.xctestSessionForWDA.start();
```

Comment:
should we reset this.xctestSessionForWDA if this call or any call below throws an exception?

Reply:
Does reset means `this.xctestSessionForWDA = null` for example? We can re-call `.start` unless the target udid hasn't changed. The caller, in xcuitest driver, calls `stop` in the case, so the `this.xctestSessionForWDA.stop` will be called in such exception case